### PR TITLE
refactor: move functions to toolkits directory

### LINKED ISSUE #610: [Feature Request] Move functions into toolkits folder
### Required prerequisites

- [X] I have searched the [Issue Tracker](https://github.com/camel-ai/camel/issues) and [Discussions](https://github.com/camel-ai/camel/discussions) that this hasn't already been reported. (+1 or comment there if it has.)
- [ ] Consider asking first in a [Discussion](https://github.com/camel-ai/camel/discussions/new).

### Motivation

we have introduced toolkits to handle with functions need to be initlized, it's better to move all functions into toolkits so that user can get tools/functions from one source.

### Solution

_No response_

### Alternatives

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by andrei3131**
## Description

Refactor package structure by moving functions to toolkits folder to enable importing toolkits and functions from one place.

## Motivation and Context

Why is this change required? What problem does it solve?
`close #610`

- [ ] I have raised an issue to propose this change ([required](https://github.com/camel-ai/camel/issues) for new features and bug fixes)

## Types of changes

What types of changes does your code introduce? Put an `x` in all the boxes that apply:

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds core functionality)
- [x] Refactor
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation (update in the documentation)
- [ ] Example (update in the folder of example)

## Implemented Tasks

- [x] Move functions to toolkits folder
- [x]  Run linter and test

## Checklist

Go over all the following points, and put an `x` in all the boxes that apply.
If you are unsure about any of these, don't hesitate to ask. We are here to help!

- [x] I have read the [CONTRIBUTION](https://github.com/camel-ai/camel/blob/master/CONTRIBUTING.md) guide. (**required**)
- [x] My change requires a change to the documentation.
- [x] I have updated the tests accordingly. (*required for a bug fix or a new feature*)
- [x] I have updated the documentation accordingly.

### CODE REVIEW

File: test/functions/test_google_maps_function.py [hunk 1]
```diff
@@ -14,11 +14,7 @@
 from functools import wraps
 from unittest.mock import MagicMock, patch
 
-from camel.functions.google_maps_function import (
-    get_address_description,
-    get_elevation,
-    get_timezone,
-)
+from camel.toolkits.google_maps_toolkit import GoogleMapsToolkit
```
Comment (Wendong-Fan) [inline test/functions/test_google_maps_function.py:17]:
```suggestion
from camel.toolkits import GoogleMapsToolkit
```

File: camel/toolkits/twitter_toolkit.py [hunk 2]
```diff
@@ -0,0 +1,524 @@
...
+
+import requests
+
+from camel.toolkits import OpenAIFunction
+from camel.toolkits.base import BaseToolkit
+
+TWEET_TEXT_LIMIT = 280
+
+
+def get_twitter_api_key() -> Tuple[str, str]:
```
Comment (Wendong-Fan) [inline camel/toolkits/twitter_toolkit.py:28]: nit: just to unify the code style as what you did for `class WeatherToolkit`, we can also move this function within the class, use `def _get_twitter_api_key` representing it's internal function

File: camel/toolkits/twitter_toolkit.py [hunk 3]
```diff
@@ -0,0 +1,524 @@
...
+            ]
+        ).strip(", ")
+        raise ValueError(
+            f"{missing_keys} not found in environment variables. Get them "
+            "here: `https://developer.twitter.com/en/portal/products/free`."
+        )
+    return TWITTER_CONSUMER_KEY, TWITTER_CONSUMER_SECRET
+
+
+def get_oauth_session() -> requests.Session:
```
Comment (Wendong-Fan) [inline camel/toolkits/twitter_toolkit.py:59]: same as above

File: camel/toolkits/twitter_toolkit.py [hunk 4]
```diff
@@ -0,0 +1,524 @@
...
+    oauth = OAuth1Session(
+        consumer_key,
+        client_secret=consumer_secret,
+        resource_owner_key=oauth_tokens["oauth_token"],
+        resource_owner_secret=oauth_tokens["oauth_token_secret"],
+    )
+    return oauth
+
+
+def handle_http_error(response: requests.Response) -> str:
```
Comment (Wendong-Fan) [inline camel/toolkits/twitter_toolkit.py:142]: same as above

**Review by Wendong-Fan — APPROVED** [review]
Thanks @andrei3131! It' great to keep both FUNCS and Toolkit for better flexibility, left some nit comments in below~

File: test/functions/test_google_maps_function.py [hunk 5]
```diff
@@ -62,7 +58,7 @@ def test_get_address_description(mock_get, mock_client):
     mock_client.return_value = mock_instance
 
     # Call the function with a test address
-    result = get_address_description(
+    result = GoogleMapsToolkit().get_address_description(
```
Comment (WHALEEYE) [inline test/functions/test_google_maps_function.py:61]: Nit: in these tests, can move the instantiation of the Toolkits into a `@pytest.fixture` annotated function.

**Review by WHALEEYE — COMMENTED** [review]
I think the tests should also be moved into `test/toolkits`?

**Review by WHALEEYE — APPROVED** [review]

**Comment by Wendong-Fan** [comment]
Thanks @andrei3131 !

**Pull request merged**